subworkflows/automate_input/bin/check_samplesheet.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python
+
+# This script is based on the example at: https://raw.githubusercontent.com/nf-core/test-datasets/viralrecon/samplesheet/samplesheet_test_illumina_amplicon.csv
+
+import os
+import sys
+import errno
+import argparse
+
+
+def parse_args(args=None):
+    Description = "Reformat nf-core/readmapping samplesheet file and check its contents."
+    Epilog = "Example usage: python check_samplesheet.py <FILE_IN> <FILE_OUT>"
+
+    parser = argparse.ArgumentParser(description=Description, epilog=Epilog)
+    parser.add_argument("FILE_IN", help="Input samplesheet file.")
+    parser.add_argument("FILE_OUT", help="Output file.")
+    return parser.parse_args(args)
+
+
+def make_dir(path):
+    if len(path) > 0:
+        try:
+            os.makedirs(path)
+        except OSError as exception:
+            if exception.errno != errno.EEXIST:
+                raise exception
```

Comment:
Can you explain the why here?

Reply:
because it's more concise ? It looks like the purpose of your try/except block is to catch cases when the directories already exist, which `makedirs` by default complains about. But with the `exist_ok` flag, those errors are not raised any more, so no need to have the try/except at al